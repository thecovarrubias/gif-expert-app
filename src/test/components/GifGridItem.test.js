import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';

import GifGridItem from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem />', () => {

    const title = 'Ejemplo'
    const url = 'http://localhost/ejemplo.jpg';

    let wrapper = shallow(
        <GifGridItem 
            title={ title } 
            url={ url }
        />
    );

    test('Debe de mostrar el componente correctamente', () => {

        expect(wrapper).toMatchSnapshot();

    });

    test('Debe de contener un párrafo con el título', () => {

        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );

    });

    test('Debe de contener los mismos props del img', () => {
        
        const img = wrapper.find('img');
        
        /* Obtener un prop y llamarlo - img.prop('src')
           Obtener todas las porps y llamar por referencia - img.props().src */

        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });

    test('Debe de contener la clase animate__fadeIn', () => {

        const div = wrapper.find('div');
        const className = div.prop('className');

        expect(className.includes('animate__fadeIn')).toBe(true);
        // expect(div.hasClass('animate__fadeIn')).toBe(true);

    });

});