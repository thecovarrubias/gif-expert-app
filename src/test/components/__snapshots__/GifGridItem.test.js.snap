// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Ejemplo"
    src="http://localhost/ejemplo.jpg"
  />
  <p>
    Ejemplo
  </p>
</div>
`;
